app/views/partials/_header.html.haml

```diff
@@ -7,6 +7,9 @@
   #navbarToggler.collapse.navbar-collapse
     %ul.navbar-nav
       %li.nav-item
-        = link_to 'Iniciar sesión', '/admin', class: 'nav-link'
+        - if user_signed_in?
+          = link_to "Sign out", users_sign_out_path, :method => :delete, class:'nav-link'
```

Comment:
Have you tried thi new Ruby sintax here, instead of `:method => :delete` now yuo might have `method: :delete` . Please try it a let me know if it works.